service/storage_service.cc

```diff
@@ -577,18 +578,14 @@ future<storage_service::nodes_to_notify_after_sync> storage_service::sync_raft_t
             tmptr->update_topology(replaced_host_id, std::nullopt, locator::node::state::being_replaced);
             update_topology(host_id, ip, rs);
             tmptr->add_replacing_endpoint(replaced_host_id, host_id);
-            if (t.tstate != service::topology::transition_state::join_group0) {
-                co_await update_topology_change_info(tmptr, ::format("replacing {}/{} by {}/{}", replaced_id, *existing_ip, id, ip));
-            } else {
-                // Do not update pending ranges in join_group0 state yet. After adding replacing endpoint above the
-                // node will no longer be reported for reads and writes, but it needs to be marked as alive
-                // before it is reported as pending. Otherwise increased CL during bootstrap will not be satisfied
-                // (replaced node cannot be contacted and replacing is reported as dead). So instead mark the node
-                // as UP (the node started report itself as alive already by this point). If it is down it will be
-                // marked as such eventually.
-                if (existing_ip == ip && !_gossiper.is_alive(*existing_ip)) {
-                    co_await _gossiper.real_mark_alive(*existing_ip);
-                }
+            co_await update_topology_change_info(tmptr, ::format("replacing {}/{} by {}/{}", replaced_id, *existing_ip, id, ip));
+            // After adding replacing endpoint above the node will no longer be reported for reads and writes,
+            // but it needs to be marked as alive before it is reported as pending. Otherwise increased CL
+            // during bootstrap will not be satisfied (replaced node cannot be contacted and replacing is reported
+            // as dead). So instead mark the node as UP (the node started report itself as alive already by this
+            // point). If it is down it will be marked as such eventually.
+            if (existing_ip == ip && !_gossiper.is_alive(*existing_ip)) {
+                co_await _gossiper.real_mark_alive(*existing_ip);
```

Comment:
We cannot just add `t.tstate == join_group0` because we cannot get here in this state. We can may be more this to else close of `if (rs.ring.has_value()) {` check. Though marking a node as alive is benign.

Reply:
I moved making alive into else clause for the if, so it will be equivalent to the previous code.